modules/caddyhttp/encode/zstd/zstd.go

```diff
@@ -38,7 +43,36 @@ func (Zstd) CaddyModule() caddy.ModuleInfo {
 }
 
 // UnmarshalCaddyfile sets up the handler from Caddyfile tokens.
-func (z *Zstd) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
+func (g *Zstd) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
+	d.Next() // consume option name
+	if !d.NextArg() {
+		return nil
+	}
+	levelStr := d.Val()
+	level, err := strconv.Atoi(levelStr)
+	if err != nil {
+		return err
+	}
+	g.Level = zstd.EncoderLevel(level)
+	return nil
+}
+
+// Provision provisions g's configuration.
+func (g *Zstd) Provision(ctx caddy.Context) error {
+	if g.Level == 0 {
+		g.Level = zstd.SpeedDefault
+	}
+	return nil
+}
+
+// Validate validates g's configuration.
+func (g Zstd) Validate() error {
+	if g.Level < zstd.SpeedFastest {
```

Comment:
Although their values are from 1 to 4, we shouldn't rely on this behaviour. They are defined as [iota](https://github.com/klauspost/compress/blob/v1.17.7/zstd/encoder_options.go#L148).

Reply:
The reason why gzip can check for number range is because the level is given explicitly as a number. But zstd only provides 4 constants.

Reply:
How would you suggest we do it then @weidideng?

Reply:
You are mostly right, but let me some explaination. I mean usage of text constants not matches caddyfile's code-style and other encoder algo's compression levels (`1-9` for `gzip` and `1-11` for `brotli`).

I think about another realization, and propose for appeal to original `zstd` levels, not library approximation. It is [1 to 22](https://github.com/facebook/zstd/blob/8689633fdf5dabfc7f4cd6f79611335c3f8def88/lib/zstd.h#L82C11-L82C12) (or [1 to 19, excludes --ultra levels](https://github.com/facebook/zstd/blob/8689633fdf5dabfc7f4cd6f79611335c3f8def88/contrib/pzstd/Options.cpp#L61) which recommended to carefull usage).

But if we implement [library's way to transform](https://github.com/klauspost/compress/blob/1e2b27500c4eae851ba753ad116dc5a6524d2366/zstd/encoder_options.go#L188) caddyfile's levels to `zstd` levels, we will have the option to specify the level 6,7,8,9 and it will be the same value (for the current version of the used library). And it will surprise the user and it will seem that changing the level does not work (in some ranges). 

Perhaps such a solution will be more universal for the long term, but now it will introduce a little confusion and crutchiness in the configuration. And even more then little: if I specify level 9 today, I will get a conditionally `N%` CPU load, then if the library implements level 9 to more powerfull compression, perhaps the CPU load will increase by a multiple, for example `3 x N%`, which may be an unpleasant surprise for a user who has updated the caddy in the future.

This is reason, why i chose to offer a version with 4 levels...

Finally, i think the best solution would be allow user to specify levels from 1 to 22 (refer to original [Zstandart levels](https://github.com/facebook/zstd/blob/8689633fdf5dabfc7f4cd6f79611335c3f8def88/lib/zstd.h#L82C11-L82C12)), convert it with [library's method](https://github.com/klauspost/compress/blob/1e2b27500c4eae851ba753ad116dc5a6524d2366/zstd/encoder_options.go#L188) and may be we may need to specify in the documentation (or even caddy warning logs) that the recommended and future safe operating `zstd` levels should be 1, 3, 6 and 10 now.

WDYT?

Reply:
I think we can use the strings/named options for zstd, I don't think that's weird. We can simply document them. We use named options in many places in the Caddyfile.

Reply:
I pushed [new commit](https://github.com/caddyserver/caddy/pull/6140/commits/0a91062a5a169243a039ba6d19a72d240915dc20) now, but my tests show error when level is not set in CaddyFile:
```
//Test results (with default 128kb zstd buffer) of 585kb minified uncompressed CSS-file
zstd (level not set) → ERR_HTTP2_PROTOCOL_ERROR (Chrome v122 browser error)
zstd fastest → 337kb 
zstd default → 294kb
zstd better → 290kb
zstd best → 286kb
```
Please friends suggest how to fix 😳

Reply:
Probably need to set a default in `Provision`